R/AbstractGraphReporter.R

```diff
@@ -94,11 +94,19 @@ AbstractGraphReporter <- R6::R6Class(
             )
 
             # Round the double columns to three digits for formatting reasons
-            numCols <- names(which(unlist(lapply(tableObj$x$data, is.double))))
+
+            # what are the double columns (but still may contain values with all integers)?
+            numDCols <- names(which(unlist(lapply(tableObj$x$data, is.double))))
+
+            # Check if each individual value in a double column is an integer,
+            # If at least one of the values is not an integers, then we want to round only those columns
+            numNonIntDCols <- names(which(unlist(lapply(tableObj$x$data[ ,numDCols],
+              function(x) sum(as.integer(x) != x) > 0))))
+
```

Comment:
Are these changes here still necessary, now that we've figured out how to generate integers at the source? I think it would now be cleaner just to check whether columns are numeric but not integer, and apply `DT::formatRound` just to those columns.

Reply:
No these are no longer necessary. My git has just gotten a bit rusty and I'm not sure whether its best to remove these commits or can you simply ignore these changes since it is in a separate file (or some other action for that matter.)

Reply:
Try something like `git checkout master -- R/AbstractGraphReporter`. That will pull in the copy of the file from the master branch. Then you can commit that. Or alternatively, you can download the file from GitHub and manually drop it into your repo. 

No worries about trying to remove the old commits, we'll squash them when we merge. 

(If you did want to try to roll back the old commits, the easiest way to do that would be to do something like `git reset --soft HEAD~4` to roll back 4 commits but keep all your changes in your workspace. Then you could reset the files you want to no longer change.)

Reply:
Thanks! I ended up copying the file and pasting it into tableformat branch. Hopefully my latest commit should resolve any formatting issues.

Reply:
I think this is still much more complicated than it needs to be. Since the vector is of mode "integer", you shouldn't need to check each individual element.

In fact, I think all you need is something like:
```
doubleCols <- names(which(unlist(lapply(tableObj$x$data, is.double))))
```

See below reprex for how the checks evaluate:

```r
intvec <- c(foo=1.0, bar=2.0)
mode(intvec) <- "integer"
is.numeric(intvec)
#> [1] TRUE
is.double(intvec)
#> [1] FALSE
is.integer(intvec)
#> [1] TRUE

doublevec <- c(foo=1.0, bar=2.0)
is.numeric(doublevec)
#> [1] TRUE
is.double(doublevec)
#> [1] TRUE
is.integer(doublevec)
#> [1] FALSE
```

<sup>Created on 2020-01-14 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

Reply:
I think you can ignore these particular changes since this was my original approach.

Reply:
Unfortunately, that is not how git works. Your changes are still in the HEAD commit of the branch you are requesting to merge in this PR. You can see the changes both in the [diff tab](https://github.com/uptake/pkgnet/pull/249/files#diff-b4b63152c558c6c3aefc3139521704f1R97-R104) for this PR and in the [source code](https://github.com/j450h1/pkgnet/blob/tableformat/R/AbstractGraphReporter.R#L96-L104) for the branch in your fork. You need to push a new commit with additional changes. I recommend you drop in 

```r
doubleCols <- names(which(unlist(lapply(tableObj$x$data, is.double))))
```

which also has the benefit of renaming that variable from `numCols` to `doubleCols` which is more descriptive and clear.  Don't forget to also change the name on [line 107](https://github.com/j450h1/pkgnet/blob/5ec33ee58ee7c46349c010b878b5401bd3cbe8a9/R/AbstractGraphReporter.R#L107) in the `DT::formatRound` call.